R/powerEmp.R

```diff
@@ -0,0 +1,155 @@
+#' based on a pre-specified critical value.
+#'
+#' @param data (`data.frame`)\cr data frame containing entry and exit times of an
+#' illness-death model. See [getSimulatedData()] for details.
+#' @param typeEvent (`string`)\cr endpoint to be evaluated, possible values are `PFS` and `OS`.
+#' @param critical (positive `number`)\cr critical value of the log-rank test.
+#'
+#' @return Logical value indicating log-rank test significance.
+#' @export
+#'
+#' @examples
+#' library(survival)
+#' transition1 <- exponential_transition(h01 = 0.06, h02 = 0.3, h12 = 0.3)
+#' transition2 <- exponential_transition(h01 = 0.1, h02 = 0.4, h12 = 0.3)
+#' simTrial <- getClinicalTrials(
+#'   nRep = 1, nPat = c(800, 800), seed = 1234, datType = "1rowPatient",
+#'   transitionByArm = list(transition1, transition2), dropout = list(rate = 0.5, time = 12),
+#'   accrual = list(param = "intensity", value = 7)
+#' )[[1]]
+#' logRankTest(data = simTrial, typeEvent = "OS", critical = 3.4)
+logRankTest <- function(data, typeEvent = c("PFS", "OS"), critical) {
+  # Must be have the format of one row per patient (`datType` must be 1rowPatient`
+  # in getClinicalTrials()), i.e. have 10 (if censored) or 11 columns.
+  assert_data_frame(data, min.cols = 10, max.cols = 11)
+  typeEvent <- match.arg(typeEvent)
+  assert_positive_number(critical)
+
+  if (typeEvent == "OS") {
+    time <- data$OStime
+    event <- data$OSevent
+  } else if (typeEvent == "PFS") {
+    time <- data$PFStime
+    event <- data$PFSevent
+  }
+
+  logRank <- survival::survdiff(Surv(time, event) ~ trt, data)
+  sqrt(logRank$chisq) > critical
+}
+
+
+#' Helper function to conduct log-rank tests for either PFS or OS
+#'
+#' @param simTrials (`list`)\cr simulated trial data sets, see [getClinicalTrials()].
+#' @param typeEvent (`string`)\cr endpoint to be evaluated, possible values are `PFS` and `OS`.
+#' @param eventNum (`integer`)\cr number of events required to trigger analysis.
+#' @param critical (positive `number`)\cr critical value of the log-rank test.
+#'
+#' @return Logical vector indicating log-rank test significance for each trial.
+#' @export
+#'
+#' @examples
+#' library(survival)
+#' transition1 <- exponential_transition(h01 = 0.06, h02 = 0.3, h12 = 0.3)
+#' transition2 <- exponential_transition(h01 = 0.1, h02 = 0.4, h12 = 0.3)
+#' simTrials <- getClinicalTrials(
+#'   nRep = 50, nPat = c(800, 800), seed = 1234, datType = "1rowPatient",
+#'   transitionByArm = list(transition1, transition2), dropout = list(rate = 0.5, time = 12),
+#'   accrual = list(param = "intensity", value = 7)
+#' )
+#' passedLogRank(simTrials = simTrials, typeEvent = "PFS", eventNum = 300, critical = 2.4)
+passedLogRank <- function(simTrials, typeEvent, eventNum, critical) {
+  assert_list(simTrials, null.ok = FALSE)
+  assert_positive_number(critical)
+  assert_count(eventNum, positive = TRUE)
+
+  # Censor simulated trials at time-point of OS/PFS analysis.
+  trialsAna <- lapply(
```

Comment:
Here and also below lapply: it would be better to name the arguments, I.e. use X and FUN for the lapply arguments and also name the arguments you pass to the FUN